v2/reader.go

```diff
@@ -121,3 +124,62 @@ func ReadVersion(r io.Reader) (version uint64, err error) {
 	}
 	return header.Version, nil
 }
+
+var _ io.ReaderAt = (*readSeekerAt)(nil)
+
+type readSeekerAt struct {
```

Comment:
should this live in `internal/io`?

Reply:
This reader is not reused elsewhere. Happy to move it there if you think it might get reused elsewhere.

Reply:
we have 3 different io interfaces we change between: `Reader`, `ReaderAt`, and `ReadSeeker`. having the internal conversions between them all in one place seems reasonable

Reply:
https://github.com/ipld/go-car/issues/145